src/main/java/software/amazon/nio/spi/s3/S3FileSystem.java

```diff
@@ -56,14 +56,16 @@ public class S3FileSystem extends FileSystem {
 
     // private S3AsyncClient client;
     private final S3NioSpiConfiguration configuration;
+    private final Path temporaryDirectory;
 
     /**
      * Create a filesystem that represents the bucket specified by the URI
      *
      * @param provider the provider to be used with this fileSystem
      * @param config   the configuration to use; can be null to use a default configuration
+     * @param temporaryDirectory the local temporary directory for the S3 object
      */
-    S3FileSystem(S3FileSystemProvider provider, S3NioSpiConfiguration config) {
```

Comment:
To maintain backward compatibility can we keep the old constructor and add this new one as well?

Reply:
Yes, I changed it.